bgpd/bgp_mplsvpn.c

```diff
@@ -516,37 +516,44 @@ static bool sid_exist(struct bgp *bgp, const struct in6_addr *sid)
 }
 
 /*
+ * This function generates a new SID based on bgp->srv6_locator_chunks and
+ * index. The locator and generated SID are stored in arguments sid_locator
+ * and sid, respectively.
+ *
  * if index != 0: try to allocate as index-mode
  * else: try to allocate as auto-mode
  */
 static uint32_t alloc_new_sid(struct bgp *bgp, uint32_t index,
-			      struct in6_addr *sid_locator)
+			      struct in6_addr *sid_locator,
+			      struct in6_addr *sid)
 {
 	struct listnode *node;
 	struct srv6_locator_chunk *chunk;
-	struct in6_addr sid_buf;
 	bool alloced = false;
 	int label = 0;
+	uint8_t offset = 0;
 
-	if (!bgp || !sid_locator)
+	if (!bgp || !sid_locator || !sid)
 		return false;
 
 	for (ALL_LIST_ELEMENTS_RO(bgp->srv6_locator_chunks, node, chunk)) {
 		*sid_locator = chunk->prefix.prefix;
-		sid_buf = chunk->prefix.prefix;
+		*sid = chunk->prefix.prefix;
+		offset = chunk->block_bits_length + chunk->node_bits_length;
```

Comment:
double-checking: offset is only u8, all good here?

Reply:
Yes.
offset indicates the position in bit sequence of IPv6 address. That is, it can be one of the values from 0 to 127, which is less than UINT8_MAX(255).